### PR TITLE
PR for Issue 862 - Add Redis Support for Notification Service

### PR DESCRIPTION
**Pull request by andresrinivasan**
#862 

IOTech deserves all the kudos. I just merged.

Signed-off-by: Andre Srinivasan <andre@redislabs.com>

### CODE REVIEW

File: internal/support/notifications/init.go [hunk 1]
```diff
@@ -95,9 +94,6 @@ func Init(useRegistry bool) bool {
 		registerUpdates = make(chan interface{})
 		go listenForConfigChanges()
 	}
-
-	go telemetry.StartCpuUsageAverage()
```
Comment (anonymouse64) [inline internal/support/notifications/init.go]: Why was this removed here?

Comment (andresrinivasan) [reply internal/support/notifications/init.go]: My guess is the IOTech branch hadn't been rebased in a bit and I missed it in the merge. I will correct.

File: internal/pkg/db/redis/data.go [hunk 2]
```diff
@@ -243,7 +243,7 @@ func (c *Client) ReadingsByDeviceAndValueDescriptor(deviceId, valueDescriptor st
 	objects, err := getObjectsByRangeFilter(conn,
 		db.ReadingsCollection+":device:"+deviceId,
 		db.ReadingsCollection+":name:"+valueDescriptor,
-		0, limit-1)
+		0, int64(limit-1))
```
Comment (tsconn23) [inline internal/pkg/db/redis/data.go]: Is this due to a concern that the `int` type may not supply a large enough value? There's no way we're going to send back 2.1 billion readings via a REST response, let alone more than that.

File: internal/pkg/db/redis/queries.go [hunk 3]
```diff
@@ -128,7 +129,7 @@ func getObjectsByRange(conn redis.Conn, key string, start, end int) (objects [][
 // Return objects by a score from a zset
 // if limit is 0, all are returned
 // if end is negative, it is considered as positive infinity
-func getObjectsByRangeFilter(conn redis.Conn, key string, filter string, start, end int) (objects [][]byte, err error) {
+func getObjectsByRangeFilter(conn redis.Conn, key string, filter string, start, end int64) (objects [][]byte, err error) {
```
Comment (tsconn23) [inline internal/pkg/db/redis/queries.go]: See comment above. I don't know that int64 buys us anything here.

File: internal/pkg/db/redis/data.go (hunk 2 above)
Comment (andresrinivasan) [reply internal/pkg/db/redis/data.go]: I was reacting to changes that came out of the notification service support. I'll revisit and follow up.

File: internal/pkg/db/redis/queries.go (hunk 3 above)
Comment (andresrinivasan) [reply internal/pkg/db/redis/queries.go]: Ditto to the above.

File: internal/pkg/db/redis/data.go (hunk 2 above)
Comment (andresrinivasan) [reply internal/pkg/db/redis/data.go]: Looks like it was propagation from the notification interface. I've corrected to an `int` implementaiton.

File: internal/pkg/db/redis/queries.go (hunk 3 above)
Comment (andresrinivasan) [reply internal/pkg/db/redis/queries.go]: Looks like it was propagation from the notification interface. I've corrected to an `int` implementaiton.

**Comment by tsconn23** [comment]
recheck
